### PR TITLE
Add automatic `Data` field deserialization for 08-wasm structs in CLI

### PR DESCRIPTION
**Pull request by vmarkushin**
Particularly, for 10-grandpa client

NOTE: this is not the final design of the feature and, most likely, will be re-implemented on the side of 08-wasm client

## Description

closes: #XXXX

### Commit Message / Changelog Entry

```bash
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/02-client/client/cli/query.go [hunk 1]
```diff
@@ -48,6 +49,10 @@ func GetCmdQueryClientStates() *cobra.Command {
 				return err
 			}
 
+			for _, cs := range res.ClientStates {
+				maybeDecodeWasmData(clientCtx, cs.ClientState)
```
Comment (blasrodri) [inline modules/core/02-client/client/cli/query.go:53]: Wouldn't it be overriding and only the last one would be valid?

Comment (vmarkushin) [reply modules/core/02-client/client/cli/query.go:53]: No, why?

Comment (blasrodri) [reply modules/core/02-client/client/cli/query.go:53]: oh, I see the cs is new. Sorry, my bad

**Review by blasrodri — APPROVED** [review]

**Comment by vuong177** [comment]
@vmarkushin @blasrodri I think  we also need to decode client state `maybeDecodeWasmData` in the querier function: https://github.com/notional-labs/ibc-go/blob/5e2c7edb6711bbd3e5efcdab9b36a7af1d801b64/modules/core/02-client/keeper/grpc_query.go#L25 and  https://github.com/notional-labs/ibc-go/blob/5e2c7edb6711bbd3e5efcdab9b36a7af1d801b64/modules/core/02-client/keeper/grpc_query.go#L100

**Pull request merged**